### PR TITLE
Restores asResultSet[M[_]] definition

### PR DESCRIPTION
**Pull request by FPerezP**

### CODE REVIEW

File: core/src/main/scala/query/interpolator/package.scala [hunk 1]
```diff
@@ -41,44 +41,46 @@ package object interpolator {
...
+        FR: FromReader[A]): FreeS[QueryModule.Op, A] = {
+      implicit val sessionAPI: SessionAPI[QueryModule.Op] = Q.sessionAPI
+      asResultSet[QueryModule.Op](consistencyLevel).flatMap(Q.resultSetAPI.read[A](_))
+    }
 
     def asOption[A](consistencyLevel: Option[ConsistencyLevel] = None)(
         implicit Q: QueryModule[QueryModule.Op],
         FR: FromReader[A]): FreeS[QueryModule.Op, Option[A]] =
-      asResultSet(consistencyLevel).flatMap(Q.resultSetAPI.readOption[A](_))
+      asResultSet[QueryModule.Op](consistencyLevel).flatMap(Q.resultSetAPI.readOption[A](_))
```
Comment (fedefernandez) [inline core/src/main/scala/query/interpolator/package.scala:60]: Why we don't need here the `sessionAPI` like in the line 55?

File: core/src/main/scala/query/interpolator/package.scala [hunk 2]
```diff
@@ -41,44 +41,46 @@ package object interpolator {
...
+        implicit API: SessionAPI[M]): FreeS[M, Unit] =
+      asResultSet[M](consistencyLevel).map(_ => (): Unit)
 
     def as[A](consistencyLevel: Option[ConsistencyLevel] = None)(
         implicit Q: QueryModule[QueryModule.Op],
-        FR: FromReader[A]): FreeS[QueryModule.Op, A] =
-      asResultSet(consistencyLevel).flatMap(Q.resultSetAPI.read[A](_))
+        FR: FromReader[A]): FreeS[QueryModule.Op, A] = {
+      implicit val sessionAPI: SessionAPI[QueryModule.Op] = Q.sessionAPI
+      asResultSet[QueryModule.Op](consistencyLevel).flatMap(Q.resultSetAPI.read[A](_))
```
Comment (fedefernandez) [inline core/src/main/scala/query/interpolator/package.scala]: Maybe we could pass the `sessionAPI` explicitly and don't declare the val

**Review by fedefernandez — COMMENTED** [review]
Left a couple of comments, thanks

File: core/src/main/scala/query/interpolator/package.scala (hunk 2 above)
Comment (FPerezP) [reply core/src/main/scala/query/interpolator/package.scala]: This is actually not needed anymore. Forgot to remove it.

File: core/src/main/scala/query/interpolator/package.scala (hunk 1 above)
Comment (FPerezP) [reply core/src/main/scala/query/interpolator/package.scala:60]: It wasn't really needed there either, I just removed it.

**Comment by FPerezP** [comment]
Thanks, @fedefernandez! I addressed them. Could you please review it again?

**Review by fedefernandez — APPROVED** [review]
LGTM! Thanks

**Pull request merged**